File: codeEg/4. 정렬/실패율.js
function solution(N, stages) {
  let answer = [];

  for (let stage = 1; stage < N + 1; stage += 1) {
    const clear = stages.filter(x => x >= stage)
    const fail = stages.filter(x => x === stage)

    if (clear.length) {
      answer.push([fail.length / clear.length , stage])
    } else {
      answer.push([0, stage])
    }
  }
  
  return answer.sort((a,b) => (b[0]-a[0]||a[1]-b[1])).map(x => x[1])
}

File: codeEg/4. 정렬/실패율.py
def solution(N, stages):
    answer = []
    array = []
    
    n = len(stages)
    for i in range(1, N+1):
      clear = len([x for x in stages if x >= i])
      if clear > 0:
        res = len([x for x in stages if x == i]) / clear
      else:
        res = 0
      array.append([res, i])
    
    for i in sorted(array, key = lambda x: (-x[0], x[1])):
      answer.append(i[1])
    return answer

solution(5, [2, 1, 2, 6, 2, 4, 3, 3])